alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/PasswordConfigField.java

```diff
@@ -22,55 +22,68 @@
+        return new PasswordConfigField(key, label, description, true, encryptionValidation, validationFunctions);
     }
 
-    public static PasswordConfigField createRequired(final String key, final String label, final String description, final ConfigValidationFunction validationFunction) {
-        return new PasswordConfigField(key, label, description, true, validationFunction);
+    public static PasswordConfigField create(String key, String label, String description, String panel, EncryptionValidator encryptionValidation) {
+        return new PasswordConfigField(key, label, description, false, encryptionValidation);
     }
 
-    public static PasswordConfigField create(final String key, final String label, final String description, final String panel) {
-        return new PasswordConfigField(key, label, description, false);
+    public static PasswordConfigField create(String key, String label, String description, String panel, EncryptionValidator encryptionValidation, ConfigValidationFunction... validationFunctions) {
+        return new PasswordConfigField(key, label, description, false, encryptionValidation, validationFunctions);
     }
 
-    public static PasswordConfigField create(final String key, final String label, final String description, final String panel, final ConfigValidationFunction validationFunction) {
-        return new PasswordConfigField(key, label, description, false, validationFunction);
+    public static PasswordConfigField createRequired(String key, String label, String description, String panel, EncryptionValidator encryptionValidation) {
+        return new PasswordConfigField(key, label, description, true, encryptionValidation);
     }
 
-    public static PasswordConfigField createRequired(final String key, final String label, final String description, final String panel) {
-        return new PasswordConfigField(key, label, description, true);
+    public static PasswordConfigField createRequired(String key, String label, String description, String panel, EncryptionValidator encryptionValidation, ConfigValidationFunction... validationFunctions) {
+        return new PasswordConfigField(key, label, description, true, encryptionValidation, validationFunctions);
     }
 
-    public static PasswordConfigField createRequired(final String key, final String label, final String description, final String panel, final ConfigValidationFunction validationFunction) {
-        return new PasswordConfigField(key, label, description, true, validationFunction);
+    private void createValidators(ConfigValidationFunction[] validationFunctions) {
+        List<ConfigValidationFunction> validators = new ArrayList<>();
+        validators.add(encryptionValidation);
+        if (null != validationFunctions) {
+            validators.addAll(Arrays.asList(validationFunctions));
+        }
+        this.setValidationFunctions(validators.stream().collect(Collectors.toUnmodifiableList()));
```

Comment:
We have almost this exact method in most ConfigFields. Perhaps we should add a method or a new ctor param to the parent class to add a default validation function?

Reply:
Because we're using method references with the this pointer I cannot add them in the constructor.  Unless I create a small private class that implements ConfigValidationFunction.